src/libcore/iter.rs

```diff
@@ -3030,6 +3030,45 @@ pub fn repeat<T: Clone>(elt: T) -> Repeat<T> {
     Repeat{element: elt}
 }
 
+/// An iterator that yields nothing.
+#[unstable(feature="core", reason = "new addition")]
```

Comment:
Could this use a different feature name than "core" here as well? Something like `iter_empty` and `iter_once` should be fine. Also, can you tag the `impl` blocks below with `#[unstable]` as well?